File: lavafast-api/src/services/lpr/recognizePlate.js
import Tesseract from "tesseract.js";

export async function recognizePlate(imagePath) {

    const resultado = await Tesseract.recognize(
        imagePath,
        "eng",
        {
            logger: m => console.log(m)
        }
    );

    return {
        texto: resultado.data.text,
        confidence: resultado.data.confidence
    };

}

File: lavafast-lpr/services/recognize_plate.py
from paddleocr import PaddleOCR

ocr = PaddleOCR(
    use_doc_orientation_classify=False,
    use_doc_unwarping=False,
    use_textline_orientation=False,
    lang="en"
)


import re

def recognize_plate(image_path):

    resultado = ocr.predict(image_path)

    textos = resultado[0]["rec_texts"]
    scores = resultado[0]["rec_scores"]

    for texto, score in zip(textos, scores):

        texto = texto.upper().replace(" ", "")

        if re.fullmatch(r"[A-Z]{3}[0-9][A-Z][0-9]{2}", texto):

            return {
                "placa": texto,
                "confidence": round(score * 100, 2)
            }

    return {
        "placa": None,
        "confidence": 0
    }